### PR TITLE
[trainer, tests, doc] feat: monitor v1 rollout-train consistency

### PR DESCRIPTION
**Pull request by WenzheWang**
### What does this PR do?

Extend the rollout-train log-prob consistency monitoring added by #291 to the
diffusion trainer v1 path introduced later by #296.

- centralize the existing availability and bypass-mode checks in a
  trainer-facing `DataProto` helper
- use the same helper from both the legacy and v1 policy-gradient trainers
- preserve the existing `rollout_corr/*` metric names and per-timestep
  breakdown
- document that consistency monitoring is available in both trainer paths

This is a follow-up to #280. It does not change rollout correction, optimization
math, defaults, or accelerator-specific code.

### Duplicate-work check

This is not duplicating an existing PR. I checked:

- [open PRs referencing #280](https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+280+in%3Abody)
- [open PRs for v1 rollout-train consistency](https://github.com/verl-project/verl-omni/pulls?q=is%3Apr+is%3Aopen+rollout+train+consistency+trainer+v1+logprob)
- the submitter's other open PRs and their changed paths

#291 added the monitoring call to the legacy diffusion trainer. The v1 trainer
was merged afterward in #296 and currently recomputes `old_log_probs` without
emitting the same diagnostics.

### Test

- Repository-pinned `verl` commit
  `8a694930275061f52ebd538c906ef8819af56dbd`:

  ```bash
  PYTHONPATH="$PWD:$VERL_PIN_CHECKOUT" python - <<'PY'
  # Bootstrap only the lightweight package paths so the CPU-only environment
  # does not import the optional vllm-omni runtime, then run:
  # pytest.main(["-q", "tests/trainer/diffusion/test_rollout_correction_on_cpu.py"])
  PY
  ```

  Result: `21 passed`.

- `pre-commit run ruff --files <changed Python files>` — passed.
- `pre-commit run ruff-format --files <changed Python files>` — passed.
- `pre-commit run mypy --files <changed Python files>` — passed.
- All remaining targeted pre-commit hooks passed when
  `autogen-trainer-cfg` was skipped.

The full targeted pre-commit run reaches an unrelated existing drift:
regenerating `_generated_omni_trainer.yaml` changes Omni configuration fields
even on the repository-pinned `verl` commit. That generated diff was reverted
and is not part of this PR.

### Additional A800 multi-step validation

A 5-step mechanism-level trainer-path run was completed on the current head
`7886eaf`:

- 1 × NVIDIA A800 80GB PCIe; PyTorch 2.6.0+cu124; bf16 tiny actor.
- Batch size 4; four diffusion/SDE timesteps (1000/750/500/250);
  deterministic seed `20260730`.
- Every step executed forward, backward, and an SGD update through the exact
  PR versions of `PolicyGradientDiffusionTrainerV1._step_once` and the
  rollout-correction helper.
- All expected `rollout_corr/*` metrics were finite across five steps;
  gradients and parameter updates were non-zero; computed fields persisted;
  the bypass-mode negative control emitted no consistency metrics.
- Actor loss changed from `0.02955836` to `0.02403400`;
  `logprob_abs_diff_mean` ranged from `6.2499e-05` to `3.1250e-04`.

Slurm job `261484` completed with exit `0:0`. Result SHA-256:
`9f7209010690364524bfde128a90a47a5bc7929328cadd2540d02a4d5aa40f73`;
log SHA-256:
`7cea1bfcebdd71156fb990a797588b3e53033beebe6fec429d6538017e9e111b`.
The full five-step table and protocol are in
[this PR comment](https://github.com/verl-project/verl-omni/pull/319#issuecomment-5130675735).

Scope: this validates the changed v1 trainer mechanism with a synthetic tiny
actor; it is not a substitute for the repository's full
Ray/TransferQueue/vLLM-Omni diffusion E2E or project GPU CI.

### API and Usage Example

No API or configuration changes. When rollout log-probs are enabled and bypass
mode is off, both trainer implementations now emit the existing diagnostics:

```yaml
actor_rollout_ref:
  rollout:
    calculate_log_probs: true
```

Examples include `rollout_corr/logprob_abs_diff_mean`,
`rollout_corr/logprob_abs_diff_max`, and
`rollout_corr/logprob_abs_diff/ts_{t}`.

### Design & Code Changes

`compute_rollout_corr_metrics_from_batch` owns the shared trainer-side gating:
it skips bypass mode and batches without both log-prob tensors, then delegates
all numerical work to the existing tested
`compute_rollout_corr_metrics_from_logprobs`. The legacy trainer is changed only
to use this helper; trainer v1 calls it immediately after recomputing
`old_log_probs` and before optional rollout correction.

### AI assistance

OpenAI Codex assisted with repository inspection, implementation, testing, and
PR preparation. The human submitter has reviewed every changed line and can
defend the change end-to-end.

### Checklist Before Starting

- [x] Searched for similar PRs and linked the queries above.
- [x] Formatted the PR title as `[{modules}] {type}: {description}` and ran the
      repository title validator.

### Checklist Before Submitting

- [x] Read the contribution guide and `AGENTS.md`.
- [x] Added/updated documentation.
- [x] Added CPU unit tests for the shared trainer-facing behavior.
- [x] Ran the relevant Ruff, formatting, mypy, structure, documentation,
      license, naming, and compile checks.
- [x] Human submitter has reviewed every changed line.
- [ ] Full `pre-commit run --all-files` — not completed because
      `autogen-trainer-cfg` exposes the unrelated generated Omni configuration
      drift described above; the generated diff was reverted.

### CODE REVIEW

File: verl_omni/trainer/diffusion/ray_diffusion_trainer.py [hunk 1]
```diff
@@ -1035,16 +1035,12 @@ def fit(self):
 
                     assert "old_log_probs" in batch.batch, f'"old_log_prob" not in {batch.batch.keys()=}'
```
[the remark this answers is not in the document]
Comment (WenzheWang) [reply verl_omni/trainer/diffusion/ray_diffusion_trainer.py]: Fixed in 7886eaf. The assertion message now matches the checked `old_log_probs` key.

**Comment by WenzheWang** [comment]
Update: addressed Copilot's assertion-message feedback in `7886eaf`; Python syntax validation passed and the review thread is resolved. The new head is mergeable and current with `main`. No workflow runs are attached to the new head yet; the previous workflow suite required approval. @SamitHuang @wtomin @zhtmike @AndyZhou952 @knlnguyen1802, please approve or retrigger the workflows as appropriate and review when convenient.

**Review by knlnguyen1802 — APPROVED** [review]
LGTM, but could you provide some running result for several steps ?

**Review by zhtmike — APPROVED** [review]

**Comment by WenzheWang** [comment]
@knlnguyen1802 Thanks — I added a 5-step A800 trainer-path run on the current head `7886eaf`.

Protocol:
- 1 × NVIDIA A800 80GB PCIe, PyTorch 2.6.0+cu124.
- Batch size 4, four diffusion/SDE steps per sample, deterministic seed `20260730`.
- A tiny bf16 GPU actor performs forward, backward, and an SGD update on every step.
- The run executes the exact `PolicyGradientDiffusionTrainerV1._step_once` and rollout-correction helper ASTs from this PR checkout.
- Main case: bypass mode off with both rollout and recomputed old log-probs. Negative control: bypass mode on.

| step | actor loss | grad L2 | update L2 | `logprob_abs_diff_mean` | max | `k3_kl` |
| ---: | ---: | ---: | ---: | ---: | ---: | ---: |
| 1 | 0.02955836 | 0.26421006 | 0.01288883 | 6.2499e-05 | 1.0000e-04 | 2.2352e-08 |
| 2 | 0.02850522 | 0.26816658 | 0.01384476 | 1.2500e-04 | 2.0000e-04 | 0 |
| 3 | 0.02729806 | 0.28216035 | 0.01384355 | 1.8750e-04 | 3.0001e-04 | 3.7253e-08 |
| 4 | 0.02590968 | 0.29243665 | 0.01444201 | 2.5000e-04 | 4.0001e-04 | 3.7253e-08 |
| 5 | 0.02403400 | 0.29619460 | 0.01463470 | 3.1250e-04 | 5.0000e-04 | 8.1956e-08 |

Validation gates passed:
- all expected `rollout_corr/*` metrics were finite on all five steps;
- timestep breakdowns were emitted for 1000/750/500/250;
- the actor had non-zero gradients and parameter updates on every step;
- computed fields were persisted after each v1 step;
- the bypass-mode control emitted no trainer-side `rollout_corr/*` metrics.

Slurm job `261484` completed with exit `0:0` in 7 seconds. Result SHA-256: `9f7209010690364524bfde128a90a47a5bc7929328cadd2540d02a4d5aa40f73`; log SHA-256: `7cea1bfcebdd71156fb990a797588b3e53033beebe6fec429d6538017e9e111b`.

Scope note: this is a mechanism-level v1 trainer-path smoke with a synthetic tiny actor. It does not replace the repository's full Ray/TransferQueue/vLLM-Omni diffusion E2E or project GPU CI.

**Review by SamitHuang — APPROVED** [review]
LGTM
